---
layout: tutorial_slides
logo: "GTN"

title: "Trajectory analysis"
questions:
  - "What is trajectory analysis?"
  - "What are the main methods of trajectory inference?"
  - "How are the decisions about the trajectory analysis made?"
  - "What to take into account when choosing the method for your data?" 
objectives:
  - "Become familiar with the methods of trajectory inference"
  - "Learn how the algorithms produce outputs"
  - "Be able to choose the method appropriate for your specific data"
  - "Gain insight into methods currently available in Galaxy"

subtopic: writing
key_points:
  - "Trajectory analysis in pseudotime is a powerful way to get insight into the differentiation and development of cells."
  - "There are multiple methods and algorithms used in trajectory analysis and depending on the dataset, some might work better than others."
  - "Trajectory analysis is quite sensitive and thus you should always check if the output makes biological sense."
contributors:
  - wee-snufkin
requirements:
  -
    type: "internal"
    topic_name: transcriptomics
    tutorials:
        - scrna-intro

---

### What is trajectory analysis?
.footnote[[Deconinck et al. Recent advances in trajectory inference from single-cell omics data](https://doi.org/10.1016/j.coisb.2021.05.005.)]
--

Trajectory inference (TI) methods have emerged as a novel subfield within computational biology to better study the underlying dynamics of a biological process of interest, such as:

cellular development    | differentiation      | immune responses
:---:| :---:| :---:
.image-40[ ![development](../../images/scrna-casestudy-monocle/1_cell_delevopment.png) ] | .image-40[ ![differentiation](../../images/scrna-casestudy-monocle/2_differentiation.png) ] | .image-40[ ![immune](../../images/scrna-casestudy-monocle/3_immune.png) ]

--
TI allows studying how cells evolve from one cell state to another, and subsequently when and how cell fate decisions are made.

.image-40[ ![process](../../images/scrna-casestudy-monocle/4_process.png) ]
---

### Clustering, trajectory and pseudotime
.footnote[[Deconinck et al. Recent advances in trajectory inference from single-cell omics data](https://doi.org/10.1016/j.coisb.2021.05.005.)]
--

.pull-left[
**Clustering** calculates cell similarities to identify group cell types, that can be identified based on the marker genes expressed in each cluster. 

.image-75[ ![clusters](../../images/scrna-casestudy-monocle/5_clusters.png) ]

]
--
.pull-right[
**Trajectory inference** helps to understand how those cell types are related - whether cells differentiate, change in response to stimuli or over time. To infer trajectories, we need data from cells at different points along a path of differentiation. This inferred temporal dimension is known as pseudotime.   
**Pseudotime** measures the cells’ progress through the transition.

.image-60[ ![trajectory](../../images/scrna-casestudy-monocle/6_trajectories.png) ]

]

---
### Trajectory inference approaches: **Clustering-based (tree based)**

.footnote[[Deconinck et al. Recent advances in trajectory inference from single-cell omics data](https://doi.org/10.1016/j.coisb.2021.05.005.)]
--

- Tries to find stable cell states in the data, then connect these states to form a trajectory and map the cells to the formed graph structure 

--

- Clustering approaches: soft K-means, Louvain, non-negative matrix factorization, hierarchical clustering

--

- To connect the clusters to each other, usually a **minimum weight spanning tree (MST)** is constructed (connecting without any cycles and with the minimum possible total edge weight)

--

.image-25[ ![mst](../../images/scrna-casestudy-monocle/mst.png) ]
---
### Trajectory inference approaches: **Graph-based**
.footnote[[Deconinck et al. Recent advances in trajectory inference from single-cell omics data](https://doi.org/10.1016/j.coisb.2021.05.005.)]
--

- Construct a graph representation of the cells and either use graph decomposition to reveal connected and disconnected components, or graph diffusion or traversal methods to construct the trajectory topology

--

- Most methods start by building a **k-nearest neighbors (kNN) ** graph using the Euclidean distance

.image-50[ ![knn](../../images/scrna-casestudy-monocle/knn.png) ]
--

- Diffusion peudotime (DPT) calculates the probability of cells transitioning into each other using random walks from a user-provided root cell and uses the differences between these probabilities as pseudotime


---

### Trajectory inference approaches: **Manifold learning-based**
.footnote[[Deconinck et al. Recent advances in trajectory inference from single-cell omics data](https://doi.org/10.1016/j.coisb.2021.05.005.)]
--

- Based on the assumption that data lie on a low-dimentional manifold embedded in a higher-dimentional space

--

- Using principal curves and graphs to obtain smooth trajectories

--

- Usually used dimentionality reduction methods: **t-SNE** (t-distributed stochastic neighbor embedding), **UMAP** (Uniform Manifold Approximation and Projection), **PCA** (Principal component analysis), **ICA** (Independent Component Analysis), **LSA** (Latent semantic analysis)

.image-60[ ![dim red](../../images/scrna-casestudy-monocle/dim_red_methods.png) ]

---

### Extensions of trajectory inference: **RNA velocity methods**
.footnote[[Deconinck et al. Recent advances in trajectory inference from single-cell omics data](https://doi.org/10.1016/j.coisb.2021.05.005.)]
--

- RNA velocity methods estimate the future state of a single cell captured in a static snapshot by looking at the ratios between spliced mRNA, unspliced mRNA, and mRNA degradation. 

--

- scVelo and CellRank use this extra velocity information to construct a directed k-nearest neighbor graph, as a starting step for the TI method. 

--

- This has the advantage that root cell specification is not necessary, and adds directional information to the trajectory.

--

.image-25[ ![velocity](../../images/scrna-casestudy-monocle/velocity.png) ]

---
### Assumptions
--

- It would be quite difficult to analyse the sample every several seconds to see how the cells are changing. Therefore, we assume that snapshot data encompass all naïve, intermediate, and mature cell states with sufficient sampling coverage to allow the reconstruction of differentiation trajectories.    
[Sagar and Grün, 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7115822/)

--

- As different TI methods make different assumptions about the data, a first choice to make is based on which biological process is to be expected: **not all TI methods are designed to infer all kinds of biological processes.**  
[Deconinck et al., 2021](https://doi.org/10.1016/j.coisb.2021.05.005)

---
### Why not all TI methods are designed to infer all kinds of biological processes?
--

Look at the pipeline presented by [Robrecht Cannoodt et al., 2016](https://onlinelibrary.wiley.com/doi/10.1002/eji.201646347)

--

.image-100[ ![pipeline](../../images/scrna-casestudy-monocle/7_methods.jpg) ]        
--

There are multiple methods using particular algorithms, or even their combinations, so you must consider which one would be best for analysing your sample.

---

### When analysing your data, consider the following: 
--

.pull-left[  
  

- Tissue from which the cells come from   

- Branching points  

- Supervised and unsupervised learning  

- Format of the data  

- Number of cells and features  

- Computing power & running time   

]

--

.pull-right[
    
 
To help you evaluate which method would work best for your data, check out this [**awesome comparision site - dynguidelines**](https://zouter.shinyapps.io/server/), a part of a larger set of open packages for doing and interpreting trajectories called [the dynverse](https://github.com/dynverse/dynverse).     
  
![zouter](../../images/scrna-casestudy-monocle/zouter.jpg) 

]

---

### When analysing your data, consider the following: **Tissue from which the cells come from**
--

.pull-left[
    
  
  
- Gives an idea about the topology you expect and the priors you can feed the method with, as well as expected cell types and potential contaminants  

- You need to know about 75% of your data and make sure your analysis shows that, for you to then identify the 25% new information  

- Trajectory analysis is quite a sensitive method, so always check if the obtained computational results make biological sense!

]

--

.pull-right[  
  

![scheme](../../images/scrna-casestudy-monocle/8_scheme.png)  

<sub><sub><sub> Reprinted from “Heterogeneity of Murine Lymph Node Stromal Cell Subsets”, by BioRender.com (2022) </sub></sub></sub>
]

---
### When analysing your data, consider the following: **Branching points**

--

Cells can differentiate or develop in various way, so they may exhibit different topologies. 

--

![table](../../images/scrna-casestudy-monocle/table.jpg)  

---

### When analysing your data, consider the following: **Branching points**  
  
  
  
.pull-left[
<br />
<br />
- diffusion pseudotime (DPT) and branch points
 - DPT does not rely on dimension reduction and can thus detect subtle changes in high-dimensional gene expression patterns
 - DPT-based analysis proceeds by identifying branching points that occur after cells progress from a root cell through a common ‘trunk trajectory’   
 <sub>[L. Haghverdi et al. (2016)](http://dx.doi.org/10.1038/nmeth.3971)</sub>

- binning and DNB (Dynamical Network Biomarker) theory
 - order cells, according to the pseudotime, evenly segmented into nonoverlap bins with each bin containing X cells
 - categorize DNB and non-DNB markers for each bin
 - one bin contains some unassigned cells - annotated as branch bin; of the remaining bins, there are bins on the trunk, and bins for each of the branch1 and branch2  
 <sub>[Z. Chen et al. (2018)](https://ieeexplore.ieee.org/document/8386685) </sub>

]

--

.pull-right[ 
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
![branch](../../images/scrna-casestudy-monocle/branch.jpg)  

]


---

### When analysing your data, consider the following: **Supervised and unsupervised learning**
--

.pull-left[  
<br />
![root_cells](../../images/scrna-casestudy-monocle/root_cells.jpg)  

]

--

.pull-right[
<br />
<br />
- Some algorithms work fully unsupervised, ie. the user is not required to input any priors.  

- However, many of them take the ‘starting cell’, or ‘end cell’ as information that helps to infer the trajectory which best represents the actual biological processes.  

- On the other hand, priors can bias the outcome of the method, but if chosen appropriately, they are really helpful.  

]

---
### When analysing your data, consider the following: **Supervised and unsupervised learning**

If you know which cells are root cells, you should enter this inforamtion to the method to make the computations more precise. However, some methods use unsupervised algorithms, so you will get a trajectory based on the tools they use and topology they can infer.    

--

Unsupervised | Priors needed: start cells | Priors needed: end cells | Priors needed: both start and end cells 
 :---:| :---:| :---: | :---: 
Slingshot, SCORPIUS, Angle, MST, Waterfall, TSCAN, SLICE, pCreode, SCUBA, RaceID/StemID, Monocle DDRTree | PAGA Tree, PAGA, Wanderlust, Wishbone, topslam, URD, CellRouter, SLICER | MFA, GrandPrix, GPfates, MERLoT | Monocle ICA

---
### When analysing your data, consider the following: **Format of the data**
--

- You have to check that your chosen method is compatible with the format of your data. If not, consider converting it or even contacting the developers to implement this into the pipeline.  

--

- Other possible inputs: 
 - Raw (FASTQ); 
 - MMformat (output of 10x); 
 - EBI-SCXA; 
 - AnnData   
   
--

- Powerful [SCEasy tool](https://toolshed.g2.bx.psu.edu/repository/display_tool?repository_id=a4381cd7162476d9&render_repository_actions_for=tool_shed&tool_config=%2Fsrv%2Ftoolshed%2Fmain%2Fvar%2Fdata%2Frepos%2F004%2Frepo_4830%2Fsceasy_convert.xml&changeset_revision=f62bc418173f) in Galaxy to convert between formats

--

- Implementations  

.image-100[ ![implementation](../../images/scrna-casestudy-monocle/implementation_1.jpg) ]

---
### When analysing your data, consider the following: **Number of cells and features**
--

- The more cells and features you have, the more dimensions there are in your dataset. Therefore, it may be more computationally difficult for the particular method to reduce the dimensionality and infer the correct trajectory.   

--

- For example, PAGA generally performs better than RaceID or Monocle when the dataset contains huge number of cells. (PAGA is graph-based, Monocle is tree based).  

--

- Also, the more cells you have, the more insightful data you can get, but it is often associated with more noise as well. Therefore, the number of cells also impacts how you pre-process your data and how ‘pure’ it will be for trajectory analysis. 

---
### When analysing your data, consider the following: **Computing power & running time**

It doesn't directly affect your analysis, however do bear in mind that calculations performed during dimensionality reduction, especially on large datasets, can be really time-consuming. Therefore, you might consider if you won't be limited by any of those factors.

---
### Trajectory analysis methods used in Galaxy
--

- PAGA ([Inferring Trajectories using Python (Jupyter Notebook) in Galaxy]({% link topics/transcriptomics/tutorials/scrna-case_JUPYTER-trajectories/tutorial.md %}))   

--

- RaceID ([Downstream Single-cell RNA analysis with RaceID]({% link topics/transcriptomics/tutorials/scrna-raceid/tutorial.md %}))  

--

- Monocle3 ([Trajectory Analysis using Monocle3]({% link topics/transcriptomics/tutorials/scrna-case_monocle3-trajectories/tutorial.md %})) 

--

- scVelo (coming soon) 

---
### Trajectory analysis methods used in Galaxy: **PAGA (Partition-based graph abstraction)**
.footnote[[F.A. Wolf et al. 'PAGA: graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells' (2019)](https://doi.org/10.1186/s13059-019-1663-x)]

--

-  Provides an interpretable graph-like map of the arising data manifold, based on estimating connectivity of manifold partitions. The graph nodes correspond to cell groups and whose edge weights quantify the connectivity between groups  

--

- By following high-confidence paths in the abstracted graph and ordering cells within each group in the path according to their distance from a progenitor cell, we trace gene changes at single-cell resolution. Hence, PAGA covers both aspects of clustering and pseudotemporal ordering by providing a coordinate system that allows us to explore variation in data while preserving its topology

--

- Provides a natural way of abstracting both topological information and information about RNA velocity  

--

- Available within [Scanpy](https://scanpy.readthedocs.io/en/stable/generated/scanpy.tl.paga.html)  

--

.image-50[ ![paga](../../images/scrna-casestudy-monocle/paga.jpg) ]

---
### Trajectory analysis methods used in Galaxy: **RaceID**
.footnote[[D. Grün et al., 'Single-cell messenger RNA sequencing reveals rare intestinal cell types' (2015)](https://doi.org/10.1038/nature14966)]

--

-  An algorithm for rare cell type identification in complex populations of single cells 

--

- The clustering step of RaceID identifies larger clusters of different cells by k-means clustering which is applied to the similarity matrix using the Euclidean metric

--

- An initial clustering is followed by an outlier identification based on a backgorund model of combined technical and biological variability in single-cell RNA-seq data obtained by quantification with unique molecular identifiers   

--

.image-50[ ![raceID](../../images/scrna-casestudy-monocle/raceid.jpg) ]

---
### Trajectory analysis methods used in Galaxy: **Monocle3**
.footnote[[C. Trapnell lab website, with detailed Monocle3 description and publications](https://cole-trapnell-lab.github.io/monocle3/)]

--

- Monocle introduced the concept of pseudotime, which is a measure of how far a cell has moved through biological progress  

--

- Orders cells by learning an explicit principal graph from the single cell genomics data with advanced machine learning techniques (Reversed Graph Embedding), which robustly and accurately resolves complicated biological processes  

--

- Performs clustering (i.e. using t-SNE and density peaks clustering), differential gene expression testing, identifies key marker genes to discover and characterise cell types, infers trajectories

--


![monocle](../../images/scrna-casestudy-monocle/monocle.jpg)  

---
### Trajectory analysis methods used in Galaxy: **scVelo**
.footnote[[ Bergen et al., 'Generalizing RNA velocity to transient cell states through dynamical modeling' (2020)](https://doi.org/10.1038/s41587-020-0591-3)]

--

- Estimates RNA velocity to study cellular dynamics, reaction rates of transcription, splicing and degradation     

--

- Identifies putative driver genes and regimes of regulatory changes and uses statistical tests, e.g., to detect different kinetics regimes     

--

- Infers a latent time to reconstruct the temporal sequence of transcriptomic events     

--

- Compatible with [scanpy](https://scvelo.readthedocs.io/) and hosts efficient implementations of all RNA velocity models    

--


.image-25[ ![scvelo](../../images/scrna-casestudy-monocle/scvelo.png) ]

---
